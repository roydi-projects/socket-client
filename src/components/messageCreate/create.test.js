import React from 'react';
import MessageCreate from '../messageCreate';
import renderer from 'react-test-renderer';
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    if (key === 'avatar') {
      return 'https://spotim-demo-chat-server.herokuapp.com/avatars/004-jigglypuff.png';
    }

    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
};

global.localStorage = new LocalStorageMock;

test('Should pass a Component snapshot test', () => {
  const sendMessage = null;
  const onSetUsername = null;
  const component = renderer.create(
    <MessageCreate sendMessage={sendMessage} onSetUsername={onSetUsername}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should not send the message', () => {
  const sendMessageMock = jest.fn();
  const onSetUsername = null;
  const component = mount(
    <MessageCreate sendMessage={sendMessageMock} onSetUsername={onSetUsername}/>,
  );
  component.instance().setState({message: ''});
  const button = component.find('[data-test="test-btn"]').hostNodes().simulate('click');
  expect(sendMessageMock.mock.calls.length).toEqual(0);
});

// test('Should send the message', () => {
//   const sendMessageMock = jest.fn();
//   const onSetUsername = null;
//   const component = mount(
//     <MessageCreate sendMessage={sendMessageMock} onSetUsername={onSetUsername}/>,
//   );
//   component.instance().setState({message: 'test'});
//   const button = component.find('[data-test="test-btn"]').hostNodes().simulate('click');
//   expect(sendMessageMock.mock.calls.length).toEqual(1);
// });
