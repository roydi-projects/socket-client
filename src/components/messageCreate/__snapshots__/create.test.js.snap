// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should pass a Component snapshot test 1`] = `
<div
  className="ui container message-create-container"
>
  <div
    className="ui grid"
  >
    <div
      className="row"
    >
      <div
        className="two wide column"
      >
        <img
          alt="https://spotim-demo-chat-server.herokuapp.com/avatars/004-jigglypuff.png"
          className="avatar"
          src="https://spotim-demo-chat-server.herokuapp.com/avatars/004-jigglypuff.png"
        />
      </div>
      <div
        className="one wide column"
      >
        <p
          className="username"
        >
          
          :
        </p>
      </div>
      <div
        className="ten wide column"
      >
        <div
          className="sc-bdVaJa cFNnwJ"
        >
          <textarea
            onChange={[Function]}
            onInput={[Function]}
            placeholder="Add a comment..."
            rows={3}
            style={
              Object {
                "resize": "",
              }
            }
            value=""
          />
        </div>
      </div>
      <div
        className="two wide column"
      >
        <button
          className="ui primary button"
          data-test="test-btn"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Send
        </button>
      </div>
    </div>
  </div>
</div>
`;
